v2/tools/generator/internal/config/object_model_configuration.go

```diff
@@ -50,6 +50,7 @@ type ObjectModelConfiguration struct {
 	ARMReference                   propertyAccess[bool]
 	ImportConfigMapMode            propertyAccess[ImportConfigMapMode]
 	IsSecret                       propertyAccess[bool]
+	PropertyNameInNextVersion      propertyAccess[string]
```

Comment:
Do we need documentation for how to use this property in azure-arm.yaml?

Reply:
We do - and we have:

``` yaml
# $nameInNextVersion: <string>
#     Gives the name this property has in the next version of the resource
#     Establishes a connection between the two properties allowing for proper
#     forward and backward conversion.
```

Reply:
I think this stems from the fact this feature was half completed - we had the configuration, but never used it.